reactor-netty-http/src/main/java/reactor/netty/http/client/HttpClientOperations.java

```diff
@@ -837,7 +850,7 @@ protected void onInboundNext(ChannelHandlerContext ctx, Object msg) {
 						httpMessageLogFactory().debug(HttpMessageArgProviderFactory.create(response)));
 			}
 
-			if (notRedirected(response)) {
+			if (notRedirected(response) && notAuthenticated()) {
```

Comment:
May be we can use another name?

```suggestion
			if (notRedirected(response) && authenticationNotRequired()) {
```